src/main/java/org/opentripplanner/standalone/server/OTPServer.java

```diff
@@ -70,4 +61,19 @@ protected void configure() {
             }
         };
     }
+
+    /**
+     * Hook to cleanup various stuff of some used libraries (org.geotools), which depend on the
+     * external client to call them for cleaning-up.
+     */
+    private static void cleanupWebapp() {
+        LOG.info("Web application shutdown: cleaning various stuff");
+        WeakCollectionCleaner.DEFAULT.exit();
+        DeferredAuthorityFactory.exit();
+    }
```

Comment:
Not sure if this is needed or not, we have run without it for a long time. I added it back because:
 - Either we use it, or
 - we delete the code.